C error building eventlog.c

MacOS, BugSur:
```
~/ocaml>gcc --version
gcc (Homebrew GCC 9.1.0) 9.1.0
```
Note, ./configure guessed the wrong compiler. On a Mac, clang should always be chosen, full stop.
I will rebuild with clang. But for now:
```
gcc -c -O2 -fno-strict-aliasing -fwrapv -Wall -Wdeclaration-after-statement -Werror -fno-common -fexcess-precision=standard -fno-tree-vrp -g  -D_FILE_OFFSET_BITS=64 -D_REENTRANT -DCAML_NAME_SPACE   -DCAMLDLLIMPORT= -DCAML_INSTR  -o eventlog.bi.o eventlog.c
eventlog.c: In function 'time_counter':
eventlog.c:108:3: error: ISO C90 forbids mixed declarations and code [-Werror=declaration-after-statement]
  108 |   uint64_t now = mach_absolute_time ();
      |   ^~~~~~~~
cc1: all warnings being treated as errors
make[2]: *** [eventlog.bi.o] Error 1
make[1]: *** [coldstart] Error 2
make: *** [world.opt] Error 2
```
If C99 is required, the configure script should have bugged out I think.

--- a/Changes
+++ b/Changes
@@ -128,6 +128,9 @@ Working version
   (Xavier Leroy, report by Github user quakerquickoats, review by
    Jeremie Dimino)
 
+- #10147, #10148: Fix building runtime with GCC on macOS.
+  (David Allsopp, report by John Skaller)
+
 OCaml 4.12.0
 ------------
 

--- a/runtime/eventlog.c
+++ b/runtime/eventlog.c
@@ -96,6 +96,7 @@ static int64_t time_counter(void)
 
 #elif defined(HAS_MACH_ABSOLUTE_TIME)
   static mach_timebase_info_data_t time_base = {0};
+  uint64_t now;
 
   if (time_base.denom == 0) {
     if (mach_timebase_info (&time_base) != KERN_SUCCESS)
@@ -105,7 +106,7 @@ static int64_t time_counter(void)
       return 0;
   }
 
-  uint64_t now = mach_absolute_time ();
+  now = mach_absolute_time ();
   return (int64_t)((now * time_base.numer) / time_base.denom);
 
 #elif defined(HAS_POSIX_MONOTONIC_CLOCK)

